Sources/MachOKit/DyldCache.swift

```diff
@@ -185,16 +185,34 @@ extension DyldCache {
 }
 
 extension DyldCache {
+    public var mainCache: DyldCache? {
+        if url.lastPathComponent.contains(".") {
+            var url = url
+            url.deletePathExtension()
```

Comment:
I have also implemented such an implementation in `MachOObjCSection` and others, 
but I noticed that it was not possible to get the main cache correctly in the following case
```
dyld_sim_shared_cache_arm64.01.dylddata
↓
dyld_sim_shared_cache_arm64.01
```


```suggestion
            let url = url
                .deletingPathExtension()
                .deletingPathExtension()
```